### PR TITLE
add additional waveform and detector library examples

### PR DESCRIPTION
**Pull request by ahnitz**
This adds an addition waveform example showing how to select higher order modes and also adds in the custom waveform example to the library waveform page as well. 

This also adds in an example of adding a custom detector in the library examples, both through the api and through a config file.

### CODE REVIEW

**Comment by ahnitz** [comment]
The changes are visible here 

https://www.atlas.aei.uni-hannover.de/work/ahnitz/doctest/html/detector.html

File: docs/index.rst [hunk 1]
```diff
@@ -16,6 +16,38 @@ gravitational-wave astronomy, please consider `contributing <https://github.com/
...
 library and application suite.
 
+.. carousel::
+    :show_controls:
+    :show_indicators:
+    :show_dark:
+    :show_shadows:
+    :show_captions_below:
+
+    .. figure:: https://collincapano.com/wp-content/uploads/2020/02/posterior3d-gw150914_masses-e1581860222179.png
```
Comment (spxiwh) [inline docs/index.rst:26]: Mild preference for having this hosted under pycbc.org.

File: examples/detector/ant.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 from pycbc.detector import Detector
 from pycbc.waveform import get_td_waveform
+from matplotlib import pyplot as plt
+import numpy as np
```
Comment (spxiwh) [inline examples/detector/ant.py:4]: Might be missing something, but don't think these new imports are used??

File: examples/waveform/higher_modes.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import matplotlib.pyplot as pp
+from pycbc.waveform import get_td_waveform
+
+# Let's plot what our new waveform looks like
+pp.figure()
+
+# You can select sets of modes or individual modes using the 'mode_array'
+for mode_select in [None,  "22 33", "21", "22", "33", "32", "44",]:
```
Comment (spxiwh) [inline examples/waveform/higher_modes.py:8]: Not sure I like the format here. I think the "proper" style for mode_array is a list, but I think we allowed weird strings to let things like config files work. Should it not be `[None, [(2,2),(3,3)] ...` here (or whatever format is expected for this after the string is parsed)?

File: docs/index.rst [hunk 4]
```diff
@@ -16,6 +16,38 @@ gravitational-wave astronomy, please consider `contributing <https://github.com/
...
+        :height: 400px
+        :target: waveform.html
+
+        Your interface to generating gravitational wave signals
+        
+    .. figure:: https://pycbc.org/pycbc/latest/html/_images/data.png
+        :height: 400px
+        :target: catalog.html
+
+        Working with gravitatoinal wave data
```
Comment (spxiwh) [inline docs/index.rst:48]: Typo: gravitational

File: docs/index.rst [hunk 5]
```diff
@@ -16,6 +16,38 @@ gravitational-wave astronomy, please consider `contributing <https://github.com/
 providing feedback, examples, documentation or helping to improve the core
 library and application suite.
 
+.. carousel::
+    :show_controls:
+    :show_indicators:
+    :show_dark:
+    :show_shadows:
+    :show_captions_below:
+
```
Comment (spxiwh) [inline docs/index.rst:25]: Suggest a reordering here. PyCBC's initial purpose was finding GW signals and working with GW data, inference came later. I would prefer to highlight the last two items *first* in the carousel (Preferred order would be [4,3,1,2] compared to current order).

**Review by spxiwh — APPROVED** [review]
Minor changes requested, happy to merge when ready. Thanks for doing this!

File: examples/detector/ant.py (hunk 2 above)
Comment (ahnitz) [reply examples/detector/ant.py:4]: Woops, good catch. I had experimented with adding some plots, but ending up not adding them this time.

File: docs/index.rst (hunk 5 above)
Comment (ahnitz) [reply docs/index.rst:25]: Sounds good, reordered as suggested.  The general idea is it should hopefully be easy for people to add more 'highlight' items here and provide a more friendly visual. I didn't really think through the order carefully. It should also autoplay after a bit.

**Pull request merged**